lib/tree.rb

```diff
@@ -16,40 +16,87 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log n)
+  # Space Complexity: O(n) - call stack
   def add(key, value)
-    raise NotImplementedError
+    if @root.nil?
+      @root = TreeNode.new(key, value)
+      return true
+    end
+
+    add_helper(key, value, @root)
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
-  def find(key)
-    raise NotImplementedError
+  def add_helper(key, value, current)
+    return TreeNode.new(key, value) if current.nil?
+
+    if key < current.key
+      current.left = add_helper(key, value, current.left)
+    else
+      current.right = add_helper(key, value, current.right)
+    end
+
+    return current
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
-  def inorder
-    raise NotImplementedError
+  # Time Complexity: O(log n)
+  # Space Complexity: O(n) - call stack
+  def find(key, current = @root)
```

Comment:
👍 The time & space complexities are both O(log n) if the tree is balanced and O(n) if unbalanced.